models/rollup_economics/ethereum/rollup_economics_ethereum_l1_fees.sql

```diff
@@ -0,0 +1,136 @@
+    incremental_strategy = 'merge',
+    unique_key = ['tx_hash'],
+    post_hook='{{ expose_spells(\'["ethereum"]\',
+                                    "project",
+                                    "rollup_economics",
+                                    \'["niftytable"]\') }}'
+)}}
+
+WITH tx_batch_appends AS (
+    SELECT
+    'Arbitrum' AS name,
+    t.block_time,
+    t.block_number,
+    t.hash,
+    (cast(gas_used as double) * (cast(gas_price as double) / 1e18)) AS gas_spent,
+    length(t.data) AS data_length
+    FROM
+    (
+      SELECT
+      evt_tx_hash AS tx_hash,
+      evt_block_time AS block_time,
+      evt_block_number AS block_number
+      FROM {{ source('arbitrum_ethereum', 'SequencerInbox_evt_SequencerBatchDeliveredFromOrigin') }} o
+      WHERE evt_block_time >= timestamp '2022-01-01'
+      {% if is_incremental() %}
+          and evt_block_time >= date_trunc('day', now() - interval '7' day)
+      {% endif %}
+
+      UNION ALL
+
+      SELECT
+      call_tx_hash AS tx_hash,
+      call_block_time AS block_time,
+      call_block_number AS block_number
+      FROM {{ source('arbitrum_ethereum','SequencerInbox_call_addSequencerL2BatchFromOrigin') }} o
+      WHERE call_success = true
+      AND call_tx_hash NOT IN
+      (SELECT evt_tx_hash FROM {{ source('arbitrum_ethereum', 'SequencerInbox_evt_SequencerBatchDeliveredFromOrigin') }} o
+      WHERE evt_block_time >= timestamp '2022-01-01'
+      )
+      {% if is_incremental() %}
+          and call_block_time >= date_trunc('day', now() - interval '7' day)
+      {% endif %}
+
+      UNION ALL
+
+      SELECT
+      call_tx_hash AS tx_hash,
+      call_block_time AS block_time,
+      call_block_number AS block_number
+      FROM {{ source('arbitrum_ethereum','SequencerInbox_call_addSequencerL2Batch') }} o
+      WHERE call_success = true
+      AND call_tx_hash NOT IN
+      (SELECT evt_tx_hash FROM {{ source('arbitrum_ethereum', 'SequencerInbox_evt_SequencerBatchDeliveredFromOrigin') }} o
+      WHERE evt_block_time >= timestamp '2022-01-01'
+      )
+      {% if is_incremental() %}
+          and call_block_time >= date_trunc('day', now() - interval '7' day)
+      {% endif %}
+
+      UNION ALL
+
+      SELECT
+      call_tx_hash AS tx_hash,
+      call_block_time AS block_time,
+      call_block_number AS block_number
+      FROM {{ source('arbitrum_ethereum','SequencerInbox_call_addSequencerL2BatchFromOriginWithGasRefunder') }} o
+      WHERE call_success = true
+      AND call_tx_hash NOT IN
+      (SELECT evt_tx_hash FROM {{ source('arbitrum_ethereum', 'SequencerInbox_evt_SequencerBatchDeliveredFromOrigin') }} o
+      WHERE evt_block_time >= timestamp '2022-01-01'
+      )
+      {% if is_incremental() %}
+          and call_block_time >= date_trunc('day', now() - interval '7' day)
+      {% endif %}
+    )b
+    INNER JOIN {{ source('ethereum','transactions') }} t
+    ON b.tx_hash = t.hash
+    WHERE b.block_number = t.block_number
```

Comment:
can you change `WHERE` to `AND` here, so that all of these are join conditions?

Reply:
will do. for my own personal learning: 

I can guess why making 'b.block_number = t.block_number' a JOIN condition rather than an additional WHERE condition is more efficient.

For a condition like 't.block_time >= timestamp '2022-01-01', where youre not comparing rows from the two tables, does putting it in the JOIN statement make youre query faster than if you made it an additional WHERE statement?